cli/azd/pkg/pipeline/pipeline_manager.go

```diff
@@ -357,7 +358,21 @@ func (pm *PipelineManager) Configure(ctx context.Context) (result *PipelineConfi
 	// Adding environment.AzdInitialEnvironmentConfigName as a secret to the pipeline as the base configuration for
 	// whenever a new environment is created. This means loading the local environment config into a pipeline secret which
 	// azd will use to restore the the config on CI
-	localEnvConfig, err := json.Marshal(pm.env.Config.Raw())
+	rawConfig := pm.env.Config.Raw()
```

Comment:
Can you add a #TODO tracker to get this moved into config itself.

Reply:
moved it as part of this PR.